src/index.js

```diff
@@ -45,61 +45,79 @@ function main(opts) {
       includeStatusCode: true,
       normalizePath: main.normalizePath,
       formatStatusCode: main.normalizeStatusCode,
-      promClient: {}
+      promClient: {},
     },
     opts
   );
   if (arguments[2] && arguments[1] && arguments[1].send) {
-    arguments[1].status(500)
-      .send('<h1>500 Error</h1>\n'
-        + '<p>Unexpected 3rd param in express-prom-bundle.\n'
-        + '<p>Did you just put express-prom-bundle into app.use '
-        + 'without calling it as a function first?');
+    arguments[1]
+      .status(500)
+      .send(
+        '<h1>500 Error</h1>\n'
+          + '<p>Unexpected 3rd param in express-prom-bundle.\n'
+          + '<p>Did you just put express-prom-bundle into app.use '
+          + 'without calling it as a function first?'
+      );
     return;
   }
 
   if (opts.prefix || opts.keepDefaultMetrics !== undefined) {
     throw new Error(
       'express-prom-bundle detected obsolete options:'
-      + 'prefix and/or keepDefaultMetrics. '
-      + 'Please refer to oficial docs. '
-      + 'Most likely you upgraded the module without necessary code changes'
+        + 'prefix and/or keepDefaultMetrics. '
+        + 'Please refer to oficial docs. '
+        + 'Most likely you upgraded the module without necessary code changes'
     );
   }
 
   if (opts.promClient.collectDefaultMetrics) {
     promClient.collectDefaultMetrics(opts.promClient.collectDefaultMetrics);
   }
 
-  const httpMetricName = opts.httpDurationMetricName || 'http_request_duration_seconds';
+  const httpDurationMetricName = opts.httpDurationMetricName || 'http_request_duration_seconds';
+  const httpBucketMetricName = opts.httpBucketMetricName || 'http_request_buckets_seconds';
 
-  const metricTemplates = {
-    'up': () => new promClient.Gauge({
-      name: 'up',
-      help: '1 = up, 0 = not up'
-    }),
-    [httpMetricName]: () => {
-      const labels = ['status_code'];
-      if (opts.includeMethod) {
-        labels.push('method');
-      }
-      if (opts.includePath) {
-        labels.push('path');
-      }
-      if (opts.customLabels){
-        labels.push.apply(labels, Object.keys(opts.customLabels));
-      }
-      const metric = new promClient.Histogram({
-        name: httpMetricName,
-        help: 'duration histogram of http responses labeled with: ' + labels.join(', '),
-        labelNames: labels,
-        buckets: opts.buckets || [0.003, 0.03, 0.1, 0.3, 1.5, 10]
-      });
-      return metric;
+  const createMetricTemplates = () => {
+    const labels = ['status_code'];
+    if (opts.includeMethod) {
+      labels.push('method');
+    }
+    if (opts.includePath) {
+      labels.push('path');
     }
+    if (opts.customLabels) {
+      labels.push.apply(labels, Object.keys(opts.customLabels));
+    }
+
+    return {
+      up: () =>
+        new promClient.Gauge({
+          name: 'up',
+          help: '1 = up, 0 = not up',
+        }),
+      [httpBucketMetricName]: () => {
```

Comment:
The split to have both is influenced by https://github.com/roylines/node-epimetheus/blob/master/lib/metrics.js.

A Summary should (by best practices) be called a `http_duration_*` metric
A Histogram should be called `http_bucket_* metric

A Histogram is used to receive metrics in buckets but is costly to be shown with percentiles on e.g. Grafana as those will be computed client side.
A Summary is preconfigured percentiles which are often shown on dashboards while the Histogram can then be used to digg deeeper.